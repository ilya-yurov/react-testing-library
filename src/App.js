import logo from './logo.svg';
import './App.css';
import { useEffect, useState } from 'react';

function App() {
	const [data, setData] = useState(null);
	const [toogle, setToogle] = useState(false);
	const [value, setValue] = useState('');

	const onClick = () => setToogle(prev => !prev)

	useEffect(() => {
		setTimeout(() => {
			setData({})
		},100)
	}, [])
	return (
		<div className="App">
			<h1 data-testid="value-elem">{value}</h1>
			{toogle === true && <div data-testid="toogle-elem">TOOGLE</div>}
			{data && <div style={{color: 'red'}}>data</div>}
			<h1>Hello world!</h1>
			<button data-testid="button-elem" onClick={onClick}>click me</button>
			<input onChange={e => setValue(e.target.value)} type="text" placeholder="input value....." />
		</div>
	);
}

export default App;
