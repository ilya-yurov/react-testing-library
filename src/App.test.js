import { render, screen, fireEvent } from '@testing-library/react';
import App from './App';

describe('React render tests', () => {
	test('Is react renders App.js', () => {
		render(<App />);
		const helloWorldElem = screen.getByText(/hello world/i);
		const button = screen.getByRole('button');
		const input = screen.getByPlaceholderText(/input value/i);
		expect(helloWorldElem).toBeInTheDocument();
		expect(button).toBeInTheDocument();
		expect(input).toMatchSnapshot();
		screen.debug();
		});

	test('async. findBy testing. style', async () => {
		render(<App />);
		const findByTextElem = await screen.findByText(/data/i)
		expect(findByTextElem).toBeInTheDocument();
		expect(findByTextElem).toHaveStyle({color: 'red'});
	});

	test('Click event testing', () => {
		render(<App />);
		const button = screen.getByTestId('button-elem');
		expect(screen.queryByTestId('toogle-elem')).toBeNull();
		fireEvent.click(button);
		expect(screen.queryByTestId('toogle-elem')).toBeInTheDocument();
	});
	test('Input testing', () => {
		render(<App />);
		const input = screen.getByPlaceholderText(/input value/i);
		expect(screen.getByTestId('value-elem')).toContainHTML('');
		fireEvent.input(input, {
			target: {value: 'Samlple text'}
		})
		expect(screen.getByTestId('value-elem')).toContainHTML('Samlple text');
		screen.debug();
	});
})

