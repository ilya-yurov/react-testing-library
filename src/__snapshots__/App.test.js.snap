// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React render tests Is react renders App.js 1`] = `
<input
  placeholder="input value....."
  type="text"
/>
`;
